lib/active_merchant/billing/gateways/redsys_rest.rb

```diff
@@ -378,6 +390,37 @@ def add_authentication(post, options)
         post[:DS_MERCHANT_MERCHANTCODE] = @options[:login]
       end
 
+      def add_stored_credentials(post, options)
+        return unless stored_credential = options[:stored_credential]
+
+        post[:DS_MERCHANT_COF_INI] = stored_credential[:initial_transaction] ? 'S' : 'N'
+
+        case stored_credential[:reason_type]
+        when 'recurring'
+          post[:DS_MERCHANT_COF_TYPE] = 'R'
+        when 'installment'
+          post[:DS_MERCHANT_COF_TYPE] = 'I'
+        else
+          post[:DS_MERCHANT_COF_TYPE] = 'C'
+        end
```

Comment:
I suggest compressing this part of code, I think it is still easy to read

```suggestion
        post[:DS_MERCHANT_COF_TYPE] = case stored_credential[:reason_type]
          when 'recurring' then 'R'
          when 'installment' then 'I'
          else
            'C'
          end
```

Reply:
💬 Comment:

just Matter of taste

```ruby
post[:DS_MERCHANT_COF_TYPE] = {
  'recurring' => 'R',
  'installment' => 'I',
  'cardholder' => 'C'
}[stored_credential[:reason_type]] || 'C'
```

Reply:
Thanks! changed!